ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -110,7 +119,58 @@ private Struct toStruct(final byte[] rowJsonBytes) {
     return struct;
   }
 
-  private Struct fromMap(final Map<String, Object> valueMap) {
+  /**
+   * Should this value be treated as a record or a top level map?
+   *
+   * <p>The deserializer supports deserializing top-level unnamed primitives, arrays and maps if
+   * the target schema only contains a single field of the required type, e.g. the value '10' can be
+   * deserialized into a schema containing a single numeric field. Likewise, the value '[1,2]' can
+   * be deserialized into a schema containing a single array field. The same is true of maps, but
+   * maps can be ambiguous.
+   *
+   * <p>There is ambiguity when handling top level maps due to the fact that JSON does not have a
+   * map type, so KSQL persists maps as JSON objects.  JSON objects are also used to represent KSQL
+   * structs. When the deserializer is presented with a JSON object and a target schema that
+   * contains a single MAP field, it is not immediately apparent if this should be deserialized as a
+   * struct or a top-level map.
+   *
+   * <p>Map fields are only ambiguous if:
+   * <ul>
+   *   <li>The target schema contains only a single field</li>
+   *   <li>and that field is a map</li>
+   *   <li>and that map has a string key</li>
```

Comment:
missing: the target schema's single field exists in the json map as a key

Reply:
done.